src/smart-components/OpenChannel/components/OpenChannelMessageList/index.tsx

```diff
@@ -64,6 +68,18 @@ function OpenchannelMessageList(
     }
   };
 
+  useEffect(() => {
+    setPageScrollHeight(scrollRef?.current?.scrollHeight || 0);
+  }, [scrollRef?.current?.scrollHeight]);
+  useLayoutEffect(() => {
+    /**
+     * The pageScrollHeight and pageScrollTop have different values
+     * with the scrollHeight and scrollTop of srollRef.current at this moment
+     */
+    const previousScrollBottom = pageScrollHeight - pageScrollTop;
+    scrollRef.current.scrollTop = scrollRef.current.scrollHeight - previousScrollBottom;
+  }, [allMessages?.length]);
+
```

Comment:
can we move this out as a custom hook?
Also, I guess we dont need a separate effect for
```
useEffect(() => {
    setPageScrollHeight(scrollRef?.current?.scrollHeight || 0);
  }, [scrollRef?.current?.scrollHeight]);
  ```

Reply:
Let me leave a comment for useEffect
> It's for saving the previous scrollHeight.
The scrollHeight values inside of useEffect and useLayoutEffect are different